bundle/scripts/scripts.go

```diff
@@ -29,11 +29,16 @@ func (m *script) Name() string {
 }
 
 func (m *script) Apply(ctx context.Context, b *bundle.Bundle) error {
-	cmd, out, err := executeHook(ctx, b, m.scriptHook)
+	executor, err := exec.NewCommandExecutor(b.Config.Path)
 	if err != nil {
 		return err
 	}
-	if cmd == nil {
+	defer executor.Cleanup()
+	wait, out, err := executeHook(ctx, executor, b, m.scriptHook)
+	if err != nil {
+		return err
+	}
+	if wait == nil {
```

Comment:
How can wait be nil if err is also nil?

I suspect that this check needs to be moved up, such that we don't execute anything if no script is defined.

Reply:
Nevermind, that is done in `executeHook`.